core/src/main/java/io/grpc/ServerCall.java

```diff
@@ -176,4 +176,16 @@ public boolean isReady() {
   public void setMessageCompression(boolean enabled) {
     // noop
   }
+
+  /**
+   * Sets the compression algorithm for this call.  If the server does not support the compression
```

Comment:
There needs to be some documentation on when this call is permitted. That can be in the form of `@throws IllegalStateException` like the other methods.